### PR TITLE
Fix item top-order logic

### LINKED ISSUE #917: Player cannot go upstairs when there is a blood splash
**Describe the bug**
When there is a blood splash on a staircase, the player is unable to climb up. The presence of the blood graphic prevents movement upstairs, blocking progression for affected players.

**To Reproduce**
1. Find or generate a blood splash on a staircase (e.g., by attacking a monster nearby).
2. Attempt to walk up the stairs as shown in the screenshot below.
3. Observe that movement upstairs is blocked.

**Expected behavior**
Players should be able to go upstairs regardless of whether a blood splash graphic is present.

**Screenshots**
Player blocked by blood splash on stairs

**Desktop (please complete the following information):**
 - OS: [e.g. Windows]
 - Client [e.g. old client, otc, otcv8]
 - Version [e.g. 8.6]

**Additional context**
This issue may affect gameplay and quest access in areas with frequent battles.

### PR DESCRIPTION
**Pull request by caioavidal**
### Description
Enhanced handling for null checks and introduced logic to manage item top-order within item operations.
fix #917

### Key Changes
- Improved null-check mechanisms to ensure robust error handling.
- Added logic to handle top-ordering of items during operations.

### Types of Changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)
- [ ] Code refactoring
- [ ] Merge Down

### Test Case
- Verify that null checks handle edge cases without errors.
- Test item operations to ensure correct top-order logic implementation.

### CODE REVIEW

File: src/Core/NeoServer.Domain/World/Structures/TileStack.cs [hunk 1]
```diff
@@ -19,12 +19,21 @@ IEnumerator IEnumerable.GetEnumerator()
     {
         return GetEnumerator();
     }
+    
+    public List<T> Values => _items;
```
[the remark this answers is not in the document]
Comment (betasantanna) [reply src/Core/NeoServer.Domain/World/Structures/TileStack.cs]: @copilot does it cause performance degradation?
